winpr/libwinpr/utils/collections/HashTable.c

```diff
@@ -148,19 +179,69 @@ static void HashTable_Rehash(wHashTable* table, int numOfBuckets)
 	table->numOfBuckets = numOfBuckets;
 }
 
-wKeyValuePair* HashTable_Get(wHashTable* table, void* key)
+static BOOL HashTable_Equals(wHashTable* table, const wKeyValuePair* pair, const void* key)
+{
+	if (!key || !pair || !table)
+		return FALSE;
```

Comment:
I feel like testing `table` here is not a good idea as for example in `HashTable_Get` we're gonna test `table` at each value of a bucket. So most probably we shall remove that test and instead test `table` at the beginning of every public `HashTable` function (I guess static private function don't need that).

Reply:
this is static, so the compiler will eliminate duplicate checks. Also updated to `INLINE`